server/src/main/java/org/opensearch/cluster/metadata/RepositoriesMetadata.java

```diff
@@ -291,7 +292,7 @@ public static void toXContent(RepositoryMetadata repository, XContentBuilder bui
             repository.cryptoMetadata().toXContent(repository.cryptoMetadata(), builder, params);
         }
         Settings settings = repository.settings();
-        if (SYSTEM_REPOSITORY_SETTING.get(settings)) {
+        if (SYSTEM_REPOSITORY_SETTING.get(settings) && params.paramAsBoolean(HIDE_SYSTEM_REPOSITORY_SETTING, false)) {
```

Comment:
Please create a backlog item to always use repositories is read from yml instead of reading from persisted cluster state. yml is always the source of truth for us.